salon_creator/lib/screens/lesson_creation_screen.dart

```diff
@@ -0,0 +1,586 @@
+          enabledBorder: OutlineInputBorder(
+            borderSide: BorderSide(
+              color: Colors.black,
+              width: 0.5,
+            ),
+          ),
+          hintText: "レッスン内容",
+        ),
+      ),
+    );
+  }
+
+  Widget _buildDeleteButton(VoidCallback function) {
+    return Container(
+      width: 40,
+      height: 40,
+      margin: EdgeInsets.all(4),
+      decoration: BoxDecoration(
+        color: primaryColor,
+        shape: BoxShape.circle,
+      ),
+      child: IconButton(
+        onPressed: function,
+        icon: Icon(
+          Icons.close,
+          color: Colors.white,
+        ),
+      ),
+    );
+  }
+
+  _buildDeleteResourceConfirmationDialog(index) {
+    return showCustomDialog(
+      context: context,
+      title: Text(
+        "本当に削除しますか?",
+        textAlign: TextAlign.center,
+        style: TextStyle(
+          fontSize: 18,
+        ),
+      ),
+      content: "選択されているファイルが削除されます",
+      leftButtonText: "削除する",
+      leftFunction: () {
+        setState(() {
+          resourcesList.removeAt(index);
+        });
+
+        Navigator.pop(context);
+      },
+      rightButtonText: "取り消し",
+      rightFunction: () {
+        Navigator.of(context).pop();
+      },
+    );
+  }
+
+  removeFile() {
+    setState(() {
+      thumbnailFile = null;
+    });
+  }
+
+  _buildDeleteThumbnailConfirmationDialog() async {
+    return showCustomDialog(
+      context: context,
+      title: Text(
+        "本当に削除しますか?",
+        textAlign: TextAlign.center,
+        style: TextStyle(
+          fontSize: 18,
+        ),
+      ),
+      content: "サムネイルを削除しますか？",
+      leftButtonText: "削除する",
+      leftFunction: () {
+        setState(() {
+          thumbnailFile = null;
+        });
+        Navigator.of(context).pop();
+      },
+      rightButtonText: "取り消し",
+      rightFunction: () {
+        Navigator.of(context).pop();
+      },
+    );
+  }
+
+  _buildDeleteMediaFileConfirmationDialog() async {
+    return showCustomDialog(
+      context: context,
+      title: Text(
+        "本当に削除しますか?",
+        textAlign: TextAlign.center,
+        style: TextStyle(
+          fontSize: 18,
+        ),
+      ),
+      content: "選択されている画像・動画が削除されます",
+      leftButtonText: "削除する",
+      leftFunction: () {
+        setState(() {
+          _mediaFile = null;
+        });
+
+        Navigator.of(context).pop();
+      },
+      rightButtonText: "取り消し",
+      rightFunction: () {
+        Navigator.of(context).pop();
+      },
+    );
+  }
+
+  Future _saveLessonDetail() async {
+    final dbHandler = DbHandler();
+    final storageHandler = StorageHandler();
+    Lesson lesson = Lesson();
+    lesson.resources = [];
+    lesson.salonId = "XytREh97Xs9pJvnI13JS";
+    lesson.name = lessonNameController?.text;
+    lesson.category = categoriesController?.text;
+    lesson.description = descriptionController?.text;
+    lesson.publish = _publish;
+    lesson.media = _mediaFile != null
```

Comment:
```suggestion
    lesson.imageUrl = _mediaFile != null
```

Reply:
This could be either a image or video. So, it's probably better to use media instead of imageUrl. But I'm not sure if it's proper to name it as media.